docs/en/query_language/functions/string_functions.md

```diff
@@ -115,13 +115,104 @@ SELECT format('{} {}', 'Hello', 'World')
 └───────────────────────────────────┘
 ```
 
-## concat(s1, s2, ...)
+## concat {#concat}
 
 Concatenates the strings listed in the arguments, without a separator.
 
-## concatAssumeInjective(s1, s2, ...)
+**Syntax** 
 
-Same as [concat](./string_functions.md#concat-s1-s2), the difference is that you need to ensure that concat(s1, s2, s3) -> s4 is injective, it will be used for optimization of GROUP BY
+```sql
+concat(s1, s2, ...)
+```
+Alias: `CONCAT`.
+
+**Parameters**
+
+- `si` – string or key. The key value can be any [data type](https://clickhouse.yandex/docs/en/data_types/).
+
+**Returned values**
+
+Returns the string that results from concatenating the arguments. 
+
+If either argument is null, `concat` returns null. 
+
+Type: `String`.
+
+**Example**
+
+Input table:
+
+```sql
+CREATE TABLE key_val(`key1` String, `key2` String, `value` UInt32) ENGINE = TinyLog
+INSERT INTO key_val VALUES ('Hello, ','World',1)('Hello, ','World',2)('Hello, ','World!',3)('Hello',', World!',2)
+SELECT * from key_val
+```
+
+```text
+┌─key1────┬─key2─────┬─value─┐
+│ Hello,  │ World    │     1 │
+│ Hello,  │ World    │     2 │
+│ Hello,  │ World!   │     3 │
+│ Hello   │ , World! │     2 │
+└─────────┴──────────┴───────┘
+```
+
+Query:
+
+```sql
+SELECT concat(key1, key2), sum(value) FROM key_val GROUP BY concat(key1, key2)
+```
+
+Result:
+
+```text
+┌─concat(key1, key2)─┬─sum(value)─┐
+│ Hello, World!      │          5 │
+│ Hello, World       │          3 │
+└────────────────────┴────────────┘
+```
+
+## concatAssumeInjective {#concatassumeinjective}
+
+Same as [concat](#concat), the difference is that you need to ensure that concat(s1, s2, ...) → sn is injective, it will be used for optimization of GROUP BY.
```

Comment:
```suggestion
Same as [concat](#concat), the difference is that you need to ensure that `concat(s1, s2, ...) → sn` is injective, it will be used for optimization of GROUP BY.
```
also we'd better link to GROUP BY chapter